### PR TITLE
Lesson 5 -- contributing

### PR DESCRIPTION
**Pull request by fmichonneau**
@mkuzak please review (or assign someone to do it).

### CODE REVIEW

File: _episodes/05-contribution-governance-communication.md [hunk 1]
```diff
@@ -1,52 +1,320 @@
...
+> - Ubuntu Contributing Guide: [short version](https://discourse.ubuntu.com/t/contribute/26) and [longer version](https://wiki.ubuntu.com/ContributeToUbuntu).
+> - [sf R package](https://github.com/r-spatial/sf#contributing)
+> - [knitr](https://github.com/yihui/knitr/blob/master/.github/CONTRIBUTING.md)
+> - [rmarkdown](https://github.com/rstudio/rmarkdown/blob/master/CONTRIBUTING.md)
+> - [Py4J - A Bridge between Python and Java](https://www.py4j.org/contributing.html)
+{: .challenge}
 
-## Communication
+As your project grows and more people start to contribute, in addition to the minimal CONTRIBUTING.md file
+you might need to provide support to your contributors for encouraging future contributions.
```
Comment (mkuzak) [inline _episodes/05-contribution-governance-communication.md]: in order to encourage

**Review by JenHarrow — COMMENTED** [review]
Typos :
line 135: reward 
Looking at the format I am wondering if this really takes 90mins especially if the people have been engaged in a little software dev before. Maybe a 60 min episode unless I have misunderstood the time needed for the challenges

**Comment by c-martinez** [comment]
## Discussion: Why allow others to contribute to your project?

Perhaps before "Why allow others to contribute to your project" I would add "How a to contribute" or rather "What is a contribution". I think it is good to clarify very early that a contribution is not necessarily a patch to your code which will suddenly fix all bugs, but that issues, suggestions and discussion are also valid and valuable forms of contribution.

## Challenge: Part II
Is there any reason to select this list of packages? It seems very R-heavy. Also, wouldn't it make sense to have the contributing guidelines for this lesson listed in there?

"You can acknowledge all contributors and reward contributions as co-authors if the contribution is substantial" -- co-authors of the software? Of future papers? Maybe it is appealing to researchers to know they can gain something by being contributors.

## Governance
Does it make sense to provide some guideline on what size of project may require what level of governance? I think @tobyhodges mentioned there was some research done on this direction (up to 150 people don't need any formal governance).

**Comment by orchid00** [comment]
This is a first milestone for the Contributions episode 
Authors @fmichonneau @Pfern @malvikasharan @orchid00 :)
and thanks to comments from @c-martinez @JenHarrow 
Please review @mkuzak @fpsom

File: _episodes/05-contribution-governance-communication.md [hunk 2]
```diff
@@ -1,52 +1,380 @@
...
 
-> ## `.checklist`
+So far you have a github repository to share your project, you've added a
+license and metadata to it. Now, to make things even better, why not think about
+inviting others to work on your project by making contributions!
+
+We can start discussing some of the aspects that you as a developer of your software
+should have in mind to make contributions as clear and transparent as possible.
+
+> ## Discussion: What are the benfits of allowing others to contribute to your project?
```
Comment (fpsom) [inline _episodes/05-contribution-governance-communication.md:29]: Typo: ben**e**fits

File: _episodes/05-contribution-governance-communication.md [hunk 3]
```diff
@@ -1,52 +1,380 @@
...
+a great idea, we can go through some questions that potential
+contributors to your open source project might want you to consider.
+
+1. Where can I contribute?
+2. How can I contribute?
+3. Who should I contact?
+4. What are the communication channels?
+5. What else should I know?
+6. What's in it for me?
+
```
Comment (fpsom) [inline _episodes/05-contribution-governance-communication.md:104]: I think there is a gap here; from the 6 questions to the single question, seems like an abrupt change. Maybe add a sentence in the form of "_Although addressing all these questions is a very good idea, sometimes people just want to focus on a single thing_".

**Review by fpsom — CHANGES_REQUESTED** [review]
LGTM, but just a few minor edits remain.

File: _episodes/05-contribution-governance-communication.md [hunk 4]
```diff
@@ -1,52 +1,380 @@
...
+  - "Define how you would support your contributors"
+  - "Establish roles, expectations and processes around contributions to the project"
+  - "Create communication strategies among people in the project"
 keypoints:
-- "Basic guidelines for how to contribute"
-- "Define comunication means"
-- "Define details like time, responsibilities, trasparency"
+  - "Having clear guidelines for contribution makes it easier for others to contribute to the project, improving it and increasing its sustainability."
+  - "Having a governance structure that is appropriate for the size of the project, will support its growth, and make it more welcoming to new contributors."
+  - "Clarifying communication means and expectations for the project saves time for current and new contributors."
```
Comment (mkuzak) [inline _episodes/05-contribution-governance-communication.md:17]: -> means of communication
what kind of expectations?

File: _episodes/05-contribution-governance-communication.md [hunk 5]
```diff
@@ -1,52 +1,380 @@
...
 ---
 
-----------------------------------------------------------------------------------------------------
 ## Clear and transparent contributions
 
-> ## `.checklist`
+So far you have a github repository to share your project, you've added a
+license and metadata to it. Now, to make things even better, why not think about
```
Comment (mkuzak) [inline _episodes/05-contribution-governance-communication.md:23]: they probably also added README.md  already

File: _episodes/05-contribution-governance-communication.md [hunk 6]
```diff
@@ -1,52 +1,380 @@
...
+inviting others to work on your project by making contributions!
```
Comment (mkuzak) [inline _episodes/05-contribution-governance-communication.md:24]: this sentence suggests we invite others by making contributions
-> inviting others to make contributions to your project

File: _episodes/05-contribution-governance-communication.md (hunk 2 above)
Comment (mkuzak) [reply _episodes/05-contribution-governance-communication.md:29]: simpler:
-> What are the benefits of the external contributions to the project?

File: _episodes/05-contribution-governance-communication.md [hunk 7]
```diff
@@ -1,52 +1,380 @@
...
+
+We can start discussing some of the aspects that you as a developer of your software
+should have in mind to make contributions as clear and transparent as possible.
+
+> ## Discussion: What are the benfits of allowing others to contribute to your project?
+> #### Time: 3 min
+> Think and list how your project might benefit from having people contributing to it.
+> What kind of challenges do you expect will arise from having outside contributors to your project?
```
Comment (mkuzak) [inline _episodes/05-contribution-governance-communication.md:32]: -> having external contributors in your project
or
-> having outsiders contributing to your project

File: _episodes/05-contribution-governance-communication.md [hunk 8]
```diff
@@ -1,52 +1,380 @@
...
+developers can improve their skills and connect people with similar
+interests while building their profile as an active contributor in the project.
+
+### Welcome your contributors!
+
+Make your contributors feel welcome, treat them well, respond to their queries
+and allow them to demonstrate their abilities. Being welcoming to your contributors
+sets the tone for future contributors, and incites future  contributors to
+also contribute to your project. To allow meaningful contributions,
+it is crucial to make your project [FAIR](https://www.nature.com/articles/sdata201618)
```
Comment (mkuzak) [inline _episodes/05-contribution-governance-communication.md:55]: how do the meaningful contributions are connected to being FAIR?

File: _episodes/05-contribution-governance-communication.md [hunk 9]
```diff
@@ -1,52 +1,380 @@
...
+useful and want to improve it. These contributors may spend considerable amount of
+their time (often their unpaid, free time) to work on your project, so be respectful.
+If your issue tracker is filled with ignored emails or comments, rude replies,
+and uncalled bad behavior, your project will not be attractive for the developers.
+
+> ## Discussion: What should others know before contributing to your project?
+> #### Time: 10 min
+> Have you ever contributed to a project? What did you need to know to
+> contribute to that project? Based on your own experience, list the minimum
+> required information someone might require to contribute to your project.
```
Comment (mkuzak) [inline _episodes/05-contribution-governance-communication.md:74]: less reapeats:
-> someone might need

File: _episodes/05-contribution-governance-communication.md [hunk 10]
```diff
@@ -1,52 +1,380 @@
...
+> > - cheer: Give a super nice welcome to contributors!
+> > - introduction: Overview of the document
+> > - how-to: Cover one or two examples of contributions and time frame
+> >
+> > References:
+> > - https://rubygarage.org/blog/how-contribute-to-open-source-projects
+> {: .solution}
+{: .discussion}
+
+If you are not yet convinced that having your project open for contributions is
```
Comment (mkuzak) [inline _episodes/05-contribution-governance-communication.md:94]: I don't know what me not being sure about the advantages of opening the project has to do with questions on how people can contribute.

File: _episodes/05-contribution-governance-communication.md [hunk 11]
```diff
@@ -1,52 +1,380 @@
...
+> #### Time: 5 minutes
+> Think about the different resources that your user might need, in
+> order to make a contribution. Also think at what stages these
+> resources might be needed.
+> Discuss in groups and list the resources.
+>
+> > ## Solution
+> >
+> > Here is a list of recommendations to provide support to your
+> > contributors, but remember this is not an extensive list.
```
Comment (mkuzak) [inline _episodes/05-contribution-governance-communication.md:181]: extensive -> exhaustive

File: _episodes/05-contribution-governance-communication.md [hunk 12]
```diff
@@ -1,52 +1,380 @@
...
+> > - Code review process
+> > - Resources to develop skills required to contribute to the project
+> > - Community calls / meetings
+> > - Archive
+> {: .solution}
+{: .discussion}
+
+### Promote your work
+
+For the developers to make relevant contributions, it is also worth thinking
```
Comment (mkuzak) [inline _episodes/05-contribution-governance-communication.md:194]: This is a very long and hard to understand sentence. Can you rewrite and simplify it?

File: _episodes/05-contribution-governance-communication.md [hunk 13]
```diff
@@ -1,52 +1,380 @@
...
+
+A properly managed project uses governance to make sure that there are no responsibilities
+left behind, unassigned, and optimizes for the rate of development progress.
+
+> ## Challenge
+> #### Time: 10 min
 >
-> Let's discuss what types of communication media do people prefer. Think about the team size, 
+> Assign levels of priority needed in the project design and governance
+> (choose one level per line and justify your choice)
```
Comment (mkuzak) [inline _episodes/05-contribution-governance-communication.md:221]: remove extra space
-> per line

File: _episodes/05-contribution-governance-communication.md [hunk 14]
```diff
@@ -1,52 +1,380 @@
...
+> > for governance and transparency in open source projects. This will clarify the contributors' roles
+> > and responsibilities within projects.
+> {: .solution}
+{: .challenge}
+
+## Communication
+
+Let's now talk about transparent communication processes. There are several options available
+to establish transparent communication with everyone involved in an open source project.
+However, the choice of communication channel differs depending on the team size, which also
```
Comment (mkuzak) [inline _episodes/05-contribution-governance-communication.md:245]: differs depending -> depends

File: _episodes/05-contribution-governance-communication.md [hunk 15]
```diff
@@ -1,52 +1,380 @@
...
+> {: .solution}
 {: .discussion}
 
-> ## `.testimonial`
-> “For me this experience has been incredible! Thank you for putting all your effort into this 
-> project. Many people responded with overflowing enthusiasm to participate and collaborate in the 
-> different facets of this project. It was a pleasure to work with all of you.” - H.S.
+## Recommendations
+
+The following checklist is only recommended key points to consider in making contributions easier, clear and transparent.
```
Comment (mkuzak) [inline _episodes/05-contribution-governance-communication.md:272]: -> the following checklist contains some points to be addressed in order to make contributions easier, clearer and more transparent

File: _episodes/05-contribution-governance-communication.md [hunk 16]
```diff
@@ -1,52 +1,380 @@
...
+> - Contributing to documentation
+> - Adding code/features
+> - Testing
+> - Writing papers
+> - Promoting
+> - Share ideas with the Community
+> - Organise events
+>
+> 2. How can I contribute?
+> - Technical requirements (software versions, space, etc)
```
Comment (mkuzak) [inline _episodes/05-contribution-governance-communication.md:289]: what kind of space? hard drive space?

File: _episodes/05-contribution-governance-communication.md [hunk 17]
```diff
@@ -1,52 +1,380 @@
...
+> - Skill requirements (programming language)
+> - Available resources
+> - Is there a style guide to refer to?
+> - What is allowed and not allowed
+> - What happens with my contributions
+> - Time frame that someone from the team responds to a contribution
+> - What if the contribution is accepted/ejected
```
Comment (mkuzak) [inline _episodes/05-contribution-governance-communication.md:296]: ejected -> rejected

File: _episodes/05-contribution-governance-communication.md [hunk 18]
```diff
@@ -1,52 +1,380 @@
...
+> - Find a mentor
+> - Teach others
+> - Learn how to work on open source projects
+>
+> 7. I don’t want to read all these, I only have a question!
+> - One contact person
+> - Minimal document
+{: .solution}
+
+## Nice examples of people contributing publicly
```
Comment (mkuzak) [inline _episodes/05-contribution-governance-communication.md:336]: those are example testimonials, not examples of people

**Review by mkuzak — CHANGES_REQUESTED** [review]

File: _episodes/05-contribution-governance-communication.md (hunk 18 above)
Comment (mkuzak) [reply _episodes/05-contribution-governance-communication.md:336]: hi all, I submitted my review comments.

**Comment by fpsom** [comment]
There are a few outstanding issues and typos in the PR. As soon as they are addressed, we can merge.

File: _episodes/05-contribution-governance-communication.md (hunk 2 above)
Comment (orchid00) [reply _episodes/05-contribution-governance-communication.md:29]: both fixed

File: _episodes/05-contribution-governance-communication.md (hunk 3 above)
Comment (orchid00) [reply _episodes/05-contribution-governance-communication.md:104]: great suggestion added and rephrased

File: _episodes/05-contribution-governance-communication.md (hunk 18 above)
Comment (orchid00) [reply _episodes/05-contribution-governance-communication.md:336]: thanks mateusz, I've address all the short reviews, will work on the longer ones with the group

**Pull request merged**